lib/fireblocks/api/request.rb

```diff
@@ -41,21 +41,24 @@ def put(body)
       valid_response!(send_request(req), request: req)
     end
 
-    def post(body)
+    def post(body, headers)
       req = Net::HTTP::Post.new(uri)
-      request_headers(body).each { |rk, rv| req[rk] = rv }
+      request_headers(body, headers).each { |rk, rv| req[rk] = rv }
+      req['Idempotency-Key'] = idempotency_key if idempotency_key
+      @idempotency_key = idempotency_key
```

Comment:
This is being set both here and L61. 

What is the purpose of setting a `idempotency_key` attr_accessor?